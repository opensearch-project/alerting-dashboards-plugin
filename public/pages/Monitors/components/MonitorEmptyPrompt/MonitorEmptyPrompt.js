/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiSmallButton, EuiEmptyPrompt, EuiText } from '@elastic/eui';

import { APP_PATH } from '../../../../utils/constants';

const filterText =
  'There are no monitors matching your applied filters. Reset your filters to view your monitors.';
const emptyMonitorText =
  'There are no existing monitors. Create a monitor to add triggers and actions.';
const loadingText = 'Loading monitors...';
const createMonitorButton = (
  <EuiSmallButton fill href={`#${APP_PATH.CREATE_MONITOR}`}>
    Create monitor
  </EuiSmallButton>
);
const resetFiltersButton = (resetFilters) => (
  <EuiSmallButton fill onClick={resetFilters}>
    Reset Filters
  </EuiSmallButton>
);

const getMessagePrompt = ({ filterIsApplied, loading }) => {
  if (loading) return loadingText;
  if (filterIsApplied) return filterText;
  return emptyMonitorText;
};

const getActions = ({ filterIsApplied, loading, resetFilters }) => {
  if (loading) return null;
  if (filterIsApplied) return resetFiltersButton(resetFilters);
  return createMonitorButton;
};

const MonitorEmptyPrompt = (props) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={
      <EuiText size="s">
        <p>{getMessagePrompt(props)}</p>
      </EuiText>
    }
    actions={getActions(props)}
  />
);

export default MonitorEmptyPrompt;
