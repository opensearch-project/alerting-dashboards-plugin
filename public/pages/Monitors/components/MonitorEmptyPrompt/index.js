/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import MonitorEmptyPrompt from './MonitorEmptyPrompt';

export default MonitorEmptyPrompt;
