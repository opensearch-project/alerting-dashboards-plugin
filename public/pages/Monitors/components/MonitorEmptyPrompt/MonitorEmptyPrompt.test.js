/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import MonitorEmptyPrompt from './MonitorEmptyPrompt';

describe('MonitorEmptyPrompt', () => {
  test('renders', () => {
    const component = <MonitorEmptyPrompt />;

    expect(render(component)).toMatchSnapshot();
  });
});
