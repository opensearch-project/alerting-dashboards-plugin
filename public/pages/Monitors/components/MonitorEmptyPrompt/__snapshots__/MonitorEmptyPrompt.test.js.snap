// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorEmptyPrompt renders 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no existing monitors. Create a monitor to add triggers and actions.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <a
    class="euiButton euiButton--primary euiButton--fill"
    href="alerting#/create-monitor"
    rel="noreferrer"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Create monitor
      </span>
    </span>
  </a>
</div>
`;
